tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/provider/JCloudBlobStoreProvider.java

```diff
@@ -0,0 +1,233 @@
+        @Override
+        public void validate(TieredStorageConfiguration config) throws IllegalArgumentException {
+            VALIDATION.validate(config);
+        }
+
+        @Override
+        public BlobStore getBlobStore(TieredStorageConfiguration config) {
+            return BLOB_STORE_BUILDER.getBlobStore(config);
+        }
+
+        @Override
+        public void buildCredentials(TieredStorageConfiguration config) {
+            if (config.getProviderCredentials() == null) {
+                try {
+                    String gcsKeyContent = Files.toString(
+                            new File(config.getConfigProperty("gcsManagedLedgerOffloadServiceAccountKeyFile")),
+                                     Charset.defaultCharset());
+                    config.setProviderCredentials(new GoogleCredentialsFromJson(gcsKeyContent).get());
+                } catch (IOException ioe) {
+                    log.error("Cannot read GCS service account credentials file: {}",
+                            config.getConfigProperty("gcsManagedLedgerOffloadServiceAccountKeyFile"));
+                    throw new IllegalArgumentException(ioe);
+                }
+            }
+        }
+    },
+
+    TRANSIENT("transient", new AnonymousProviderMetadata(new TransientApiMetadata(), "")) {
+        @Override
+        public void validate(TieredStorageConfiguration config) throws IllegalArgumentException {
+            if (Strings.isNullOrEmpty(config.getBucket())) {
+                throw new IllegalArgumentException(
+                    "Bucket cannot be empty for Local offload");
+            }
+        }
+
+        @Override
+        public BlobStore getBlobStore(TieredStorageConfiguration config) {
+            if (!BlobStoreRepository.containsKey(config.getBlobStoreLocation())) {
+                BlobStore bs = ContextBuilder.newBuilder(config.getDriver().toLowerCase())
+                        .buildView(BlobStoreContext.class)
+                        .getBlobStore();
+
+                if (!bs.containerExists(config.getBucket())) {
+                    Location loc = new LocationBuilder()
+                            .scope(LocationScope.HOST)
+                            .id(UUID.randomUUID() + "")
+                            .description("Transient " + config.getBucket())
+                            .build();
+
+                    bs.createContainerInLocation(loc, config.getBucket());
+                }
+                BlobStoreRepository.put(config.getBlobStoreLocation(), bs);
+            }
+            return BlobStoreRepository.get(config.getBlobStoreLocation());
+        }
+
+        @Override
+        public void buildCredentials(TieredStorageConfiguration config) {
+            // No-op
+        }
+    };
+
+    public static final boolean driverSupported(String driverName) {
+        for (JCloudBlobStoreProvider provider: JCloudBlobStoreProvider.values()) {
```

Comment:
S3 also needs to provide aws-s3, otherwise there's a BC break.

Reply:
I have added an S3 enumeration which enables the configuration of service endpoints that provide an S3-compatible interface as described here: https://jclouds.apache.org/guides/aws-s3/

Do we have a valid S3 endpoint to test against?